### PR TITLE
apache2_module: Accept the result of a2enmod when available

### PR DESCRIPTION
**Pull request by julien-lecomte**
##### SUMMARY

Fixes https://github.com/ansible-collections/community.general/issues/4592 with least changes for when a2enmod exists.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
apache2_module

##### ADDITIONAL INFORMATION

The use case is the following: have the module run on a setup where the apache2 configuration is invalid because sites-enabled already contains missing apache2 modules.

Without ignore_configcheck, the module fails on configuration check.

With ignore_configcheck, the module fails by ignoring the output of a2enmod and only rechecking the configuration, which can still have other missing modules.

With the patch, in the second case, a second check is done on the output of a2enmod to decide what to return. If the module has been installed, result is success+changed. If the module was already installed, result is success+unchanged.

This patch is the minimalistic correction that does not alter previous methods nor does change the check_mode method.

### CODE REVIEW

**Comment by julien-lecomte** [comment]
For the unit tests, 4 fail without any patch, 5 fail with the patch.

**Comment by felixfontein** [comment]
Continuation of #4593.

**Comment by felixfontein** [comment]
In CI all unit tests pass. What fails are some integration tests.

**Comment by julien-lecomte** [comment]
Seems to break https://github.com/ansible/ansible/issues/33669 CI tests.

File: plugins/modules/web_infrastructure/apache2_module.py [hunk 1]
```diff
@@ -218,6 +220,8 @@ def _set_state(module, state):
             module.exit_json(changed=True,
                              result=success_msg,
                              warnings=module.warnings)
+        elif result == 0:
+            module.exit_json(changed=('already' not in stdout), result=success_msg)
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/apache2_module.py:224]: Does it make sense to add spaces around `already`?
```suggestion
            module.exit_json(changed=(' already ' not in stdout), result=success_msg)
```

File: tests/integration/targets/apache2_module/tasks/actualtest.yml [hunk 2]
```diff
@@ -101,7 +101,7 @@
 
   - name: use identifier to enable module, fix for https://github.com/ansible/ansible/issues/33669
     community.general.apache2_module:
-      name: dump_io
+      name: dumpio
```
Comment (felixfontein) [inline tests/integration/targets/apache2_module/tasks/actualtest.yml:104]: The goal of this test is that it uses `dump_io` and not `dumpio`, if I understand the issue correctly. So this should not be changed.

Comment (julien-lecomte) [reply tests/integration/targets/apache2_module/tasks/actualtest.yml:104]: Hi

It was late last night, so I didn't have time to comment on this bit. I still have to write up some remarks on this whole PR and the issue trying to be solve.

This aside, a2enmod is aware of the particular naming (at least on debian bullesye, buster and stretch), and will correctly install the module:
```
(.venv) julien@dev:~/src/misc/ansible_collections/community/general$ sudo a2enmod dump_io
Enabling module dump_io.
To activate the new configuration, you need to run:
  systemctl restart apache2
```

I believe that the command `ansible-test integration  --docker --python 3.9  apache2_module` runs the latest ubuntu, and even it succeeds in installing dump_io via `a2enmod dump_io`

Maybe it's best to use a module other than dump_io to test the identifier keyword.

**Comment by julien-lecomte** [comment]
While investigating this PR, several issues have come to light:

1. `dump_io` is actually a correctly, accepted, name for `a2enmod`. The reason the test successfully failed was because of `apache2ctl -M` reporting ("dumpio_module") which is no longer relied upon exclusively. "ignore_configcheck" is not relevant in this case since it's only checked on pre-detection of existing modules, and not post-detection after a2enmod action.
2. This PR tries to solve the main issue: if the apache2 configuration is broken, then modules should still be enabled without issue as it only depends on a2enmod. This patch kinda solves this, but still requires all playbooks to have ignore_configcheck set to yes. This is subobtimal, and kinda ugly. Currently, if apache configuration is broken, the module will never be able to install a module, and this becomes a fatal error for entire playbooks.
3. Of note, a2enquery now exists on bullseye.

I'm partially satisfied in this PR, as it's a stop-gap and not a serene solution. We could always accept a finalized version of this PR in the first time, but I believe that a partial rewrite of this module might be necessary (proposing myself for rewrite). The only issue I see in a rewrite that does not depend on apache2ctl is how to do the config_check mode without altering the system (without a2enmod, and without a2enquery on previous platforms) 

cc authors of plugin for input if they wish to express their opinions @robinro, @n0trax, @berendt

File: plugins/modules/web_infrastructure/apache2_module.py (hunk 1 above)
Comment (julien-lecomte) [reply plugins/modules/web_infrastructure/apache2_module.py]: It makes sense. Committed.

**Comment by julien-lecomte** [comment]
Ping?

**Pull request closed without merging**